src/server/api/phone.js

```diff
@@ -16,12 +16,14 @@ export const GraphQLPhone = new GraphQLScalarType({
     if (ast.kind !== Kind.STRING) {
       throw new GraphQLError(
         `Query error: Can only parse strings got a: ${ast.kind}`,
-        [ast]
+        { nodes: [ast] }
```

Comment:
What does nodes: [ast] do?

Is this a new dictionary process within graphql?  With a data-type change this significant, I'm just curious how you knew this line needed to be updated.  And how you knew this was the correct format?

Reply:
A combination of https://www.apollographql.com/docs/apollo-server/data/errors/ and https://graphql-kr.github.io/graphql-js/error/

This code needs another pass though, which I'll do in a later commit.

Reply:
`nodes: [ast]` passes the abstract syntax tree for the graphql query into the error object. The particular code you highlighted looks at the `ast` and decides if  something passed as a phone number looks like a phone number and raises an exception if not. This is happening in the `parseLiteral` hook which gives us a way to inspect and possibly alter a literal during parsing. `GraphQLPhone` is a custom type in our code to handle phone numbers.

Reply:
Interesting!  So I checked out the custom type hook you mentioned.  

Confirming this is the correct file path: `src/server/api/phone.js`

The above file is imported to as GraphQLPhone inside our schema `src/server/api/schema.js` and then exported within the constant `resolvers` as a value for Phone.

I'm looking for the functionality where you say we could "possibly alter a literal."  And now I see that the parseLiteral hook is being used on line 15 of the `phone.js` file.

Would this functionality get in the way of seeing a toll-free number or a short-code?

Reply:
I do not know. This is not new functionality. It might be worth entering a ticket to test that and possibly fix it.

For contacts, it's unlikely we'll be using a toll-free number or short code to reach them. This code would be problematic if was used for the phone numbers we use to originate the texts, such as in the number purchasing code (as one example).